haystack/nodes/label_generator/pseudo_label_generator.py

```diff
@@ -60,7 +60,7 @@ class PseudoLabelGenerator(BaseComponent):
     def __init__(
         self,
         question_producer: Union[QuestionGenerator, List[Dict[str, str]]],
-        retriever: BaseRetriever,
+        retriever,
```

Comment:
Same question as above: Why can't we keep the type annotation? Is it because of circular imports? Can we find a better workaround?